server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java

```diff
@@ -0,0 +1,117 @@
+package org.infinispan.server.functional;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.marshall.MarshallerUtil;
+import org.infinispan.commons.marshall.ProtoStreamMarshaller;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
+import org.infinispan.query.api.continuous.ContinuousQuery;
+import org.infinispan.query.api.continuous.ContinuousQueryListener;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import org.infinispan.server.test.InfinispanServerRule;
+import org.infinispan.server.test.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * @author Katia Aresti &lt;karesti@redhat.com&gt;
+ * @since 10.0
+ **/
+public class HotRodCacheContinuousQueries {
```

Comment:
We have some CQ tests in old server. I'd prefer to have those moved to the new server before adding other tests.

Reply:
No, because they use entities, instead of proto.

Reply:
RemoteContinuousQueryIT uses protobuf

Reply:
And besides CQ, we need to ensure filters+converters work in general. RemoteListenerWithDslFilterIT if testing that.